R/glue.R

```diff
@@ -289,11 +289,35 @@ trim <- function(x) {
 }
 
 #' @export
-print.glue <- function(x, ..., sep = "\n") {
-  x[is.na(x)] <- style_na(x[is.na(x)])
+print.glue <- function(x, ..., sep = "\n", max = getOption("max.print")) {
+  cat("<glue[", length(x), "]>\n", sep = "")
+  if (length(x) == 0) {
+    return(invisible())
+  }
+
+  truncated <- length(x) > max
+  if (truncated) {
+    x <- x[seq_len(max)]
```

Comment:
Do we want this to affect the return value? I.e. `print()` no longer passes `x` through invisibly.